### PR TITLE
Separate unconfirmed points in all results data

### LINKED ISSUE #941: A+ All results table (course points page) should not include unconfirmed points in the normal official points
The All results table includes unconfirmed points in the normal points, but it should not. It must be possible to separate normal real points from the unconfirmed and unofficial points. It must also be possible to combine normal and unconfirmed points without unofficial points.

The table should have checkboxes for selecting
1. normal points
2. normal with unconfirmed points
3. normal with unofficial points
4. normal with unconfirmed and unofficial

In other words, it should NOT combine the options "unofficial" and "unconfirmed" into a single checkbox. It must be possible to separate unconfirmed points from unofficial.

Unconfirmed points are part of the mandatory feedback feature, which is triggered by the category setting `confirm_the_level` (Mandatory activity).

Related to #918.

### PR DESCRIPTION
**Pull request by lainets**
# Description

**What?**

Add checkbox to the all results UI to include/exclude unconfirmed points and backend logic for it.

**Why?**

#941

**How?**

Add a filter to the submission query to filter out unconfirmed submissions.

Closes #941

# Testing

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [X] Django unit tests.
- [X] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [X] Manual testing.

Made unconfirmed submissions and checked that the filter leaves them out.

**Did you test the changes in**

- [X] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

# Translation

- [X] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [X] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [X] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

### CODE REVIEW

**Comment by lainets** [comment]
This doesn't handle unconfirmed points for exercises that are direct children of a module instead of a learning object. I figure this is required, and will be adding it tomorrow.

**Comment by lainets** [comment]
This should only be merged after https://github.com/apluslms/a-plus/pull/1194.

Also, I wonder if fetching all submissions from the server and doing the computations on the client would be prohibitively slow or take too much RAM. It seems like that would be much simpler than the current solution. (also stops the need to reload and remake the results table from scratch)

File: exercise/api/csv/views.py [hunk 1]
```diff
@@ -318,17 +330,83 @@ def retrieve(self, request, version=None, course_id=None, user_id=None):
...
+                        output_field=IntegerField(),
+                    )
+                )
+                .filter(
+                    Q(parent_id=F("outer_parent_id"))
+                    # Exercises directly under a module have NULL parents
+                    | Q(parent_id=None, outer_parent_id=None),
+                )
+                .filter(
+                    course_module__course_instance = self.instance,
```
Comment (Mankro) [inline exercise/api/csv/views.py:365]: Inconsistent whitespace around `=`

**Review by Mankro — CHANGES_REQUESTED** [review]
Great work! A couple of very minor details should be cleaned up. I am not very familiar with the OuterRef query sets so I don't completely understand the new database query for the confirmed submissions. I will trust you with that. Hopefully, the new queries with the added complexity don't perform too slowly in the large courses.

File: exercise/templates/exercise/staff/results.html [hunk 2]
```diff
@@ -96,11 +96,17 @@
...
+					data-placement="right"
+					title="{% translate 'UNOFFICIAL_POINTS_TOOLTIP' %}">
+						<input type="checkbox" class="unofficial-checkbox" value="official" checked>
+						{% translate "RESULTS_SHOW_UNOFFICIAL_POINTS" %}
+				</label>
+				<label style="margin-left: 10px;" class="checkbox-inline" data-toggle="tooltip"
+					data-placement="right"
+					title="{% translate 'UNCONFIRMED_POINTS_TOOLTIP' %}">
+						<input type="checkbox" class="unconfirmed-checkbox" value="official" checked>
+						{% translate "RESULTS_SHOW_UNCONFIRMED_POINTS" %}
```
[the remark this answers is not in the document]
Comment (Mankro) [reply exercise/templates/exercise/staff/results.html:108]: About the `checked` attribute, I suppose the default value for loading the points data is really set in the JavaScript. So the visible "checked" state of the checkboxes must match the behavior of the JavaScript code. The JS does not automatically load unofficial points when the page is opened, right?

[the remark this answers is not in the document]
Comment (lainets) [reply exercise/templates/exercise/staff/results.html]: I changed the values to `unofficial` and `unconfirmed`, and set them to be unchecked initially. I also refactored the load function to load the variables from the DOM, so the event listeners don't have to.

File: exercise/static/exercise/results_staff.js [hunk 3]
```diff
@@ -1122,7 +1122,17 @@
      * Loads new data on page load, and when "Show only official points" checkbox clicked
      * @param {*} show_unofficial
      */
-    function loadStudentData(show_unofficial, ignore_last_grading_mode) {
+    function loadStudentData(show_unofficial, show_unconfirmed, ignore_last_grading_mode) {
+        if(show_unofficial == undefined) {
+            show_unofficial = $('input.unofficial-checkbox').prop('checked')
+        }
+        if(show_unconfirmed == undefined) {
+            show_unconfirmed = $('input.unconfirmed-checkbox').prop('checked')
```
Comment (Mankro) [inline exercise/static/exercise/results_staff.js:1130]: Semicolons at the end of the lines.

**Review by Mankro — APPROVED** [review]
Thanks!

**Pull request merged**